src/azure-cli/azure/cli/command_modules/batch/custom.py

```diff
@@ -73,12 +74,29 @@ def get_account(cmd, client, resource_group_name=None, account_name=None):
 @transfer_doc(AutoStorageBaseProperties)
 def create_account(client,
                    resource_group_name, account_name, location, tags=None, storage_account=None,
-                   keyvault=None, keyvault_url=None, no_wait=False):
+                   keyvault=None, keyvault_url=None, no_wait=False, public_network_access=None,
+                   encryption_key_source=None, encryption_key_identifier=None):
     properties = AutoStorageBaseProperties(storage_account_id=storage_account) \
         if storage_account else None
+    if public_network_access and public_network_access != "Enabled" and public_network_access != "Disabled":
```

Comment:
if you use arg_type=get_three_state_flag(), you can remove this check

Reply:
That forces having a default in the CLI. None is valid on the wire and manually setting a separate default in the client therefore seems invalid